Reviews table update

Sbonelo01 opened the pull request:
Related issues: [https://github.com/Umuzi-org/Tilde/issues/139]

## Description: Reviews Table Update

What are you up to? Fill us in :)

## Screenshots/videos

![Screenshot (66)](https://user-images.githubusercontent.com/37469171/129642573-a78b0ae4-e8de-4606-84d9-bfd5adef3a41.png)
![Screenshot (67)](https://user-images.githubusercontent.com/37469171/129642577-5b28c581-efde-451f-b194-fbd0f5a5a984.png)
![Screenshot (68)](https://user-images.githubusercontent.com/37469171/129642578-43dfe94c-30a7-40d1-8cda-5148eb62e440.png)


<!-- If there is a visual component to what you did, please save us time by adding a screenshot. You can even link to a video demonstrating your feature, that would be cool too -->

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

Sbonelo01 committed:
set up

Sbonelo01 committed:
table ready

Sbonelo01 committed:
mostly done - need to figure out the badges

Sbonelo01 committed:
staging

Sbonelo01 committed:
working fix

Sbonelo01 committed:
Merge branch 'reviews-table-update' of https://github.com/Umuzi-org/Tilde into reviews-table-update
This is a working update of the reviews table

Sbonelo01 committed:
left to do: sort, badges, make use of imports

Sbonelo01 committed:
unedit reviews.js and created a dummy file

sheenarbw commented on frontend/src/components/regions/CardDetailsModal/ReviewsSbonelo.js:
```diff
@@ -0,0 +1,65 @@
+  let theme = createTheme();
+  theme = responsiveFontSizes(theme);
+
+  const review =
```
These should come from props. Please dont hard-code values except if you are working in a story

Sbonelo01 replied:
This has been modified and not hardcoded

sheenarbw commented on frontend/src/components/regions/CardDetailsModal/ReviewsTable.js:
```diff
@@ -0,0 +1,21 @@
+const useStyles = makeStyles({
+  root: {
+    minWidth: false
+  },
```
please edit frontend/src/components/regions/CardDetailsModal/Reviews.js so it displays what we need. Dont make a whole new thing, rather just fix it in place.

If you have a car and you take it to the mechanic because the starter moter is broken, and then the mechanic doesn't fix anything and just gives you a new starter moter in your hand, would you be pleased?

Sbonelo01 replied:
This has been edited and takes the review component from `frontend/src/components/regions/CardDetailsModal/Review.js`

sheenarbw commented on frontend/src/components/regions/CardDetailsModal/ReviewsTable.js:
```diff
@@ -0,0 +1,21 @@
+  },
+});
+
+export default function ReviewsTable() {
```
Rule number 1 of naming things: Call it what it is.

Is this a table?

Sbonelo01 replied:
This has been fixed as well

sheenarbw commented on frontend/src/stories/ReviewsTable.stories.js:
```diff
@@ -0,0 +1,9 @@
+import React from "react";
```
We dont need a new story. We need the existing stories to draw the correct things.

Run the storybook and look at these urls:
http://localhost:6006/?path=/story/tilde-carddetailsmodal-reviews--loading
http://localhost:6006/?path=/story/tilde-carddetailsmodal-reviews--primary

These need to draw the correct thing

sheenarbw replied:
frontend/src/stories/ReviewsTable.stories.js should be deleted

Sbonelo01 replied:
The http://localhost:6006/?path=/story/tilde-carddetailsmodal-reviews--primary now draws the correct things. I have also edited the http://localhost:6006/?path=/story/tilde-carddetailsmodal-reviews--loading to show the tilde loader

sheenarbw commented on frontend/src/components/regions/CardDetailsModal/Reviews.js:
```diff
@@ -86,7 +86,7 @@ export default ({ reviewIds, reviews }) => {
           <TableHead>
             <TableRow>
-              <TableCell>Timestamp</TableCell>
+              <TableCell>timestamp</TableCell>
```
Why are you correcting this? Aren't you meant to be completely replacing this?

sheenarbw replied:
Please upgrade this file so that the existing stories draw the correct things

Sbonelo01 replied:
This has also been amended

Sbonelo01 replied:
Tables have been removed entirely

sheenarbw commented on frontend/src/components/regions/CardDetailsModal/ReviewsSbonelo.js:
```diff
@@ -0,0 +1,65 @@
+import React from "react";
```
Should this file exist? 

And if it should exist why does it have such a fully name?  Please name things properly. 

If I ask you "what does this component?" You will say "It is a card that displays a single review". So "ReviewCard" is a better name. 

This component does not draw multiple reviews. So having a pleural in the name is not appropriate.
This component is also not a Sbonelo.

If we need to see who wrote the code, we can use git.

Sbonelo01 replied:
This has definitely been removed. It was for practice, I pushed the code to save my work

sheenarbw commented on frontend/package.json:
```diff
@@ -4,16 +4,19 @@
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.54",
     "@material-ui/pickers": "^3.2.10",
+    "@material-ui/x-grid-data-generator": "^4.0.0-alpha.37",
```
You have added some new dependencies but I dont see you using them anytwhere. Are you sure it's necessary? Or it s just making the application heavier and harder to updrade with no benefit? 

Please remove the unnecessary things

Sbonelo01 replied:
I have been threading softly with trying to remove dependencies, But I will be removing them

sheenarbw replied:
Still planning to emove this:    ` "@material-ui/x-grid-data-generator": "^4.0.0-alpha.37",`

Try to keep new dependencies to a minimum. a lot of stuff just adds bloat

sheenarbw reviewed (changes requested):
There is a lot that needs to be fixed here. It looks like you aren't sure what success looks like.

The thing is that we need your code to be a part of Tilde. When the students look at the reviews thety got, then we want things to look better.

We dont need some weird thing off to the side. We need your code to be integrated into the rest of Tilde.

We have apis that fetch an appay of reviews. The review array gets passed into a component. That component needs to draw the correct thing. 

If you just make a completely seperate component and nothing relies on it, feeds it data or uses it in any way then it's not useful. The point of this excercise is not to make stories. Stories are useless until the components they demonstrate are actually used.

Commit:
show a single review card

Commit:
single review story

Sbonelo01 committed:
organizing files better

Sbonelo01 committed:
removing padding to save space

Sbonelo01 committed:
removing an extra character

babalwa-mbolekwa reviewed (changes requested):
Hi Sbonelo, so far so good, please address these changes:
- Please remove the **ReviewsSbonelo.js** file
- I suggest placing the date, then the reviewer's email, then `CardActions` before the review comments and reduce the spacing at the top and bottom of the `CardActions`.
- On the story file for Review, it would be good to give it the same title as its component (Review) instead of _SingleReview_

Sbonelo01 committed:
padding fix

Sbonelo01 committed:
removing files that are not nessessary

Sbonelo01 committed:
integrating a single review card to reviews

Sbonelo01 committed:
padding and card views fixed

Sbonelo01 commented:
> Hi Sbonelo, so far so good, please address these changes:
> 
> * Please remove the **ReviewsSbonelo.js** file
> * I suggest placing the date, then the reviewer's email, then `CardActions` before the review comments, and reduce the spacing at the top and bottom of the `CardActions`.
> * On the story file for Review, it would be good to give it the same title as its component (Review) instead of _SingleReview_

Hey Babalwa, I have removed the files, I have also swapped the date and reviewer email and removed the padding. The other suggested changes don't look aesthetically pleasing when doing the changes

babalwa-mbolekwa reviewed (changes requested):
Well done Sbonelo!
- Please also change the body content for **Loading...** and **No reviews yet** conditions since you are no longer using a table. I suggest just surrounding them with a paragraph tag.
- Please remove all unused components imports inside the  **Reviews.js** file and all other unused variables that are generating warnings in the code, like the `timestamp` variable

Sbonelo01 committed:
adding a loader for reviews and removing unused table imports

Sbonelo01 committed:
wrapping "No reviews yet" properly

Sbonelo01 committed:
import clean-up

Sbonelo01 committed:
fixing the issue of space presevation by hiding large text/comment

Sbonelo01 commented:
**I have added a space-saving mechanism**

on Desktop

![Screenshot from 2021-09-05 21-10-12](https://user-images.githubusercontent.com/37469171/132138672-6d8d421b-f846-4267-97ad-9b61dbb859d1.png)

on Mobile

![Screenshot from 2021-09-05 21-09-59](https://user-images.githubusercontent.com/37469171/132138705-323639a0-6f9b-4e61-b174-59f91cfd205a.png)

Sbonelo01 committed:
touch up on look

Sbonelo01 committed:
cleanup

Sbonelo01 committed:
removed the "data-grid-module"

Sbonelo01 committed:
clean

sheenarbw commented on frontend/package.json:
```diff
@@ -4,16 +4,18 @@
     "date-fns": "^2.14.0",
     "gray-matter": "^4.0.2",
     "javascript-time-ago": "^2.3.7",
+    "mui-datatables": "^3.7.7",
```
where is this being used?

sheenarbw reviewed (changes requested):
You still sem to have some extra dependencies . Can you please uninstall them?

You can see what you added by looking at the PR files. Please remove the stuff you added that you aren't using.

https://github.com/Umuzi-org/Tilde/pull/147/files

Sbonelo01 committed:
removed "x-grid-data-generator" & "mui-tables"

Sbonelo01 commented:
Apologies, they have been removed

sheenarbw reviewed (changes requested):
Ok, looking good.

Just some small things:

Can you put the email address and timestamp on the same line (saving some space) 
And then give the reviews a more distinctive border just so they dont look like they are all connected together?

Sbonelo01 commented:
> Ok, looking good.
> 
> Just some small things:
> 
> Can you put the email address and timestamp on the same line (saving some space)
> And then give the reviews a more distinctive border just so they dont look like they are all connected together?

Goodmorning @sheenarbw , should the time stamp come before or after the email address? Also, Would it not look better if I greyed out the header(where the email and email address are) instead of adding a border?

Sbonelo01 commented:
> Ok, looking good.
> 
> Just some small things:
> 
> Can you put the email address and timestamp on the same line (saving some space)
> And then give the reviews a more distinctive border just so they dont look like they are all connected together?

Goodmorning Sheena, In every way I have tried having the _timestamp_ and _reviewer email address_ on the same line, It ends up looking really bad on mobile view. Please advise.

sheenarbw commented:
ok, no worries then :)
